src/Controller/ContentManagement/ContentTypeController.php

```diff
@@ -1019,7 +1033,9 @@ private function treatFieldSubmit(ContentType $contentType, FieldType $field, $a
 
         if (in_array($action, ['save', 'saveAndClose'])) {
             $field->updateOrderKeys();
-            $contentType->setDirty($contentType->getEnvironment()->getManaged());
+            $env = $contentType->getEnvironment();
+            $managed = $env ?  $env->getManaged() : false;
```

Comment:
If env is false you should throw a RuntimeException('Unexpected not found environment')